src/dmd/globals.d

```diff
@@ -378,15 +380,18 @@ struct Loc
     const(char)* filename;
     uint linnum;
     uint charnum;
+    // 0-based offset from origin in bytes (allows O(1) access)
+    uint bytes;
```

Comment:
Adding this field, since Loc is attached to every AST node, will result in a very large increase in memory consumption.